tasks/install-matlab/v1/src/mpm.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright 2023 The MathWorks, Inc.
+
+import * as taskLib from "azure-pipelines-task-lib/task";
+import * as toolLib from "azure-pipelines-tool-lib/tool";
+import * as path from "path";
+import * as matlab from "./matlab";
+
+export async function setup(platform: string, architecture: string): Promise<string> {
+    const mpmRootUrl: string = "https://www.mathworks.com/mpm/";
+    let mpmUrl: string;
+    if (architecture !== "x64") {
+        return Promise.reject(Error(`This action is not supported on ${platform} runners using the ${architecture} architecture.`));
+    }
+    let mpm: string;
+    let mpmExtractedPath: string;
+    switch (platform) {
+        case "win32":
+            mpmUrl = mpmRootUrl + "win64/mpm";
+            mpm = await toolLib.downloadTool(mpmUrl);
+            mpmExtractedPath = await toolLib.extractZip(mpm);
+            mpm = path.join(mpmExtractedPath, "bin", "win64",  "mpm.exe");
+            break;
+        case "linux":
+            mpmUrl = mpmRootUrl + "glnxa64/mpm";
+            mpm = await toolLib.downloadTool(mpmUrl);
+            const exitCode = await taskLib.exec("chmod", ["+x", mpm]);

```

Comment:
This doesn't have to be done on macOS?

Reply:
Hmmm... It's not failing without it, but it was failing for Linux. We could add it for macOS as well for some peace of mind.